csvcubed/csvcubed/cli/inspect/metadataprocessor.py

```diff
@@ -0,0 +1,59 @@
+"""
+Metadata Processor
+---------------------------
+
+Provides functionality for validating and detecting input metadata.json file.
+"""
+
+import json
+import logging
+
+from pathlib import Path
+from rdflib import Graph, URIRef
+
+_TEMP_PREFIX_URI = URIRef("http://temporary")
+
+_logger = logging.getLogger(__name__)
+
+
+class MetadataProcessor:
+    """
+    This class handles the loading of metadata jsons to RDFLib Graphs.
+    """
+
+    def __init__(self, csvw_metadata_file_path: Path):
+        self.csvw_metadata_file_path = csvw_metadata_file_path
+
+    def load_json_ld_to_rdflib_graph(self) -> Graph:
+        """
+        Loads CSV-W metadata json-ld to rdflib graph
+
+        Member of :class:`./MetadataProcessor`.
+
+        :return: `Graph` - RDFLib Graph of CSV-W metadata json.
+        """
+        csvw_metadata_file_path = self.csvw_metadata_file_path.absolute()
+        csvw_rdf_graph = Graph(base=_TEMP_PREFIX_URI)
```

Comment:
The temporary prefix URI was only necessary so that I could retain relative URIs when serialising the graph back out to JSON-LD in the pmdutils project. I don't expect you'll be writing back to CSV-Ws here, so you can remove the arg.

Reply:
The temp prefix is now removed.